federated_aws_cli/cache.py

```diff
@@ -29,7 +59,54 @@ def wrapper(*args, **kwargs):
     return wrapper
 
 
-@__requires_safe_cache_dir
+@contextmanager
+def _safe_write(path):
+    # Try to open the file as 600
+    f = os.fdopen(os.open(path, os.O_WRONLY | os.O_CREAT | os.O_TRUNC, mode=0o600), "w")
+    yield f
+    f.close()
+
+
+@_requires_safe_cache_dir
+def read_group_role_map(url):
+    # Create a sha256 of the endpoint url, so fix length and remove weird chars
+    url = sha256(url.encode("utf-8")).hexdigest()
```

Comment:
A bit odd to overwrite the `url` variable with new content that is a hash instead of a url. Works but a bit confusing since, below it looks like path is set to `rolemap_` `+` `url` but actually it's `rolemap_` `+` `hashed_url`